sphinx_js/renderers.py

```diff
@@ -109,8 +110,20 @@ def _formal_params(self, doclet):
         """Return the JS function or class params, looking first to any
         explicit params written into the directive and falling back to
         those in the JS code."""
-        return (self._explicit_formal_params or
-                ('(%s)' % ', '.join(doclet['meta']['code'].get('paramnames', []))))
+        if self._explicit_formal_params:
+            return self._explicit_formal_params
+
+        # Try to use documented params (@param) first
```

Comment:
Note to self: this adds params from the @param tag iff they're not subfields of of a collision with an existing one.

Reply:
This part is maybe overcomplicated... Should I rewrote this using a `for` loop to make it easier to read?

Reply:
No, I think it's fine and probably has advantages in guaranteeing a correct termination condition; I'll just add a comment.